src/site/content/en/blog/module-workers/index.md

```diff
@@ -0,0 +1,229 @@
+---
+title: "Threading the Web with Module Workers"
+subhead: |
+  Moving heavy lifting into background threads is now easier with JS Modules in Workers.
+date: 2019-12-17
+hero: hero.jpg
+alt: Module Workers
+authors:
+  - developit
+description: |
+  JavaScript is single-threaded, but Web Workers can be used to move expensive operations into
+  background threads. Module Workers bring modern JavaScript
+tags:
+  - post
+  - web-workers
+  - js-modules
+---
+
+<!--lint disable no-heading-punctuation heading-increment -->
+
+JavaScript is single-threaded, which means only one operation can be performed at a time. This is
+intuitive and works well for lots of cases on the web, but can become problematic when we need to
+do heavy lifting tasks like data processing, parsing, computation or analysis. As more and more
+complex applications are delivered on the Web Platform, there's an increased need for
+multi-threaded processing.
+
+On the Web Platform, the main primitive for threading and parallelism is the [Web
+Worker](https://developer.mozilla.org/en-US/docs/Web/API/Web_Workers_API/Using_web_workers).
+Workers are a lightweight abstraction on top of [Operating System
+threads](https://en.wikipedia.org/wiki/Thread_%28computing%29) that expose a message passing API
+for inter-thread communication. This can be immensely useful when performing costly computations or
+operating on large datasets, allowing the main thread to run smoothly while performing the
+expensive operations on one or more background threads.
+
+Here's a typical example of Worker usage, where a worker script listens for messages from the main
+thread and responds by sending back messages of its own:
+
+
+##### page.js:
+
+```js
+const worker = new Worker('worker.js');
+worker.addEventListener(e => {
+  console.log(e.data);
+});
+worker.postMessage('hello');
+```
+
+##### worker.js:
+
+```js
+addEventListener('message', e => {
+  if (e.data === 'hello') {
+    postMessage('world');
+  }
+});
+```
+
+
+Web Workers have actually been available in most browsers for over 10 years. While that means they
+have excellent browser support and are well-optimized, it also means they long predate JavaScript
+Modules. Since there was no module system when Workers were designed, the API for loading code into
+a Worker and composing scripts has remained similar to the synchronous script loading approaches
+common in 2009.
+
+## History: Classic Workers
+
+The Worker constructor takes a [Classic
+Script](https://html.spec.whatwg.org/multipage/webappapis.html#classic-script) URL, which is
+relative to the document URL. It immediately returns a reference to the new Worker instance, which
+exposes a messaging interface as well as a `terminate()` method that immediately stops and destroys
+the worker.
+
+```js
+const worker = new Worker('worker.js');
+```
+
+An `importScripts()` function is available within Web Workers for loading additional code, but it
```

Comment:
'web workers' should actually be lower case. You would use upper case for referring to the API by name or if referring to the interface explicitly. Please check this throughout.

Reply:
There are other capitalization things I noted. I corrected them in some places but not others.